ERCS/erc-7750.md

```diff
@@ -0,0 +1,80 @@
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-08-04
+---
+
+## Abstract
+
+This ERC proposes a decentralized employment system (DES) built on the Ethereum blockchain. The system records employment history, allows users to found new companies, facilitates labor contracts, automates salary payments, and includes a review system for both companies and employees. All employment history is public, ensuring transparency and accountability in the employment process.
+
+## Motivation
+
+Traditional employment systems are centralized, opaque, and often lack trust. The DES aims to bring transparency, immutability, and trust to employment by leveraging blockchain technology. By recording employment history on-chain, enabling decentralized company creation, and automating contract enforcement, the DES promotes a fairer and more transparent employment ecosystem.
+
+## Specification
+
+### Company Creation
+
+1. **Company Registration**: Users can register new companies on the blockchain. Each company is a unique entity with its own address.
+2. **Company Profile**: The company profile includes details like name, industry, and a list of registered employees.
+
+### Labor Contracts
+
+1. **Contract Creation**: A standardized labor contract is created between the company and the employee. The contract includes terms like salary, duration, responsibilities, and termination conditions.
+2. **Contract Tokenization**: Each contract is represented as a unique token (e.g., an NFT) that is signed by both the company and the employee.
+3. **Contract Execution**: The contract is enforced by the smart contract, ensuring compliance with the agreed terms.
+
+### Employment History
+
+1. **Immutable Records**: Employment history is recorded on the blockchain, creating an immutable record of a person’s employment journey.
+2. **Public Accessibility**: All employment histories are public and can be accessed by anyone, promoting transparency.
+
+### Payment System
+
+1. **Salary Deposits**: Companies must deposit salary funds into the contract as per the agreed terms. These funds are held in escrow until payment conditions are met.
+2. **Automated Payments**: The smart contract automates salary payments according to the contract terms, ensuring timely and accurate payments.
+
+### Firing Employees
+
+1. **Termination Conditions**: The contract specifies conditions under which the company can terminate an employee. These conditions are enforced by the smart contract.
+2. **Dispute Resolution**: A decentralized dispute resolution mechanism can be integrated to handle disagreements between companies and employees.
+
+### Review System
+
+1. **Reputation Scores**: Both companies and employees can receive and give reviews. These reviews are stored on-chain and contribute to a reputation score.
+2. **Public Reviews**: Reviews are public and accessible to all users, fostering accountability and trust in the system.
```

Comment:
You're kind of describing your desired outcome without giving any specific technical instructions to implementers. How will someone know how to represent a review stored on-chain? What functions need to be called? What contracts does the system use?

This is way too vague, even for a draft.